### PR TITLE
Fix unstable drop table unit test

### LINKED ISSUE #5060: Drop table unit test is not stable
## Enhancement
The drop table unit test is not stable and may fails some time: https://ci.pingcap.net/blue/organizations/jenkins/tiflash-sanitizer-daily/detail/tiflash-sanitizer-daily/268/pipeline/

### PR DESCRIPTION
**Pull request by lidezhu**
### What problem does this PR solve?

Issue Number: close #5060

Problem Summary: To make the segments in the unit test split more easily, we change the dt related params in settings.
But we just change the settings in the local context, and in the global context the settings is unchanged.
When processing write request, `DeltaMergeStore` calls `checkSegment` with the local context we passed. So it splits as we expected.
But we do several `DeltaMergeStore` initialization in the test, and in the initialization process, `DeltaMergeStore` calls `checkSegment` with the global_context, and try to merge the segments which make the test not stable

### What is changed and how it works?
Make the settings in local context and global_context consistent.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Storages/DeltaMerge/tests/gtest_dm_storage_delta_merge.cpp [hunk 1]
```diff
@@ -717,14 +717,28 @@ CATCH
...
+    auto & global_settings = ::DB::tests::TiFlashTestEnv::getGlobalContext().getSettingsRef();
+    // store the old value to restore global_context settings after the test finish to avoid influence other tests
+    auto old_global_settings = global_settings;
+    SCOPE_EXIT({
+        global_settings = old_global_settings;
+    });
+    // change the settings to make it more easy to trigger splitting segments
+    Settings settings;
+    settings.dt_segment_limit_rows = 11;
     settings.dt_segment_limit_rows = 11;
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/tests/gtest_dm_storage_delta_merge.cpp:729]: A duplicated line in L728~L729

**Review by JaySon-Huang — APPROVED** [review]
LGTM with small comment

**Pull request merged**